packages/modules/imodel-browser/src/containers/iModelGrid/useIModelData.ts

```diff
@@ -32,153 +32,224 @@ export const useIModelData = ({
   apiOverrides,
   searchText,
   maxCount,
-  viewMode,
 }: IModelDataHookOptions) => {
+  const [needsUpdate, setNeedsUpdate] = React.useState(true);
+  const [iModels, setIModels] = React.useState<IModelFull[]>([]);
+  const [status, setStatus] = React.useState<DataStatus>();
+  const [page, setPage] = React.useState(0);
+  const [morePagesAvailable, setMorePagesAvailable] = React.useState(true);
+  const [abortController, setAbortController] = React.useState<
```

Comment:
this seems odd, do we really need to manage the abortController with useState?
why isnt this just using useMemo?

Reply:
Not sure how a useMemo would work here... It's not a value that can be recalculated based off other state changing.

Essentially, I'm trying to prevent multiple simultaneous requests. I always want to abort the previous request if a new one starts.
- If the consumer calls fetchMore before the previous fetch finishes, the previous request is still valid so I do nothing.
- If the external state (the component arguments) changes, the previous request is invalid, so we abort it and start a new one.
- If the component unmounts, we abort the ongoing request.

Reply:
Why do you need to call `setAbortController(updateAbortController);`? 
wouldnt calling updateAbortController suffice if it was just aborting the abortController from state?

Reply:
It admittedly looks stupid to separate out the setter from the update. 

I think I did it for a reason because my test fails if I combine them. Give me a few minutes to figure out if it's a stupid reason 😄

Reply:
The separation was a convoluted way to abort the active request on unmount, but not when an effect reran. 
Was stupid. I got rid of the function and made the abort controller cleanup more explicit.

Reply:
Much cleaner, thx